lib/SearchAndSort/SearchAndSort.js

```diff
@@ -987,9 +987,19 @@ class SearchAndSort extends React.Component {
     const count = source.totalCount();
     const messageKey = resultCountMessageKey || 'stripes-smart-components.searchResultsCountHeader';
 
-    const paneSub = !source.loaded()
-      ? <FormattedMessage id="stripes-smart-components.searchCriteria" />
-      : <FormattedMessage id={messageKey} values={{ count }} />;
+    // when there are > 10k results to a search, the `totalRecords`
+    // value comes back as `999999999`, so we use that value as a
+    // trigger to display an "unknown total count" value.
+    let paneSub;
+    if (source.loaded) {
+      if (count === 999999999) {
```

Comment:
Maybe it makes sense to have a constant for this magic number

Reply:
OMG yes, my CS-101 prof. would have failed an assignment if I turned in code like that! 🤦‍♂